src/status_im/contexts/wallet/send/select_address/view.cljs

```diff
@@ -151,9 +152,11 @@
         input-value    (reagent/atom "")
         input-focused? (reagent/atom false)]
     (fn []
-      (let [selected-tab          (or (rf/sub [:wallet/send-tab]) (:id (first tabs-data)))
-            valid-ens-or-address? (boolean (rf/sub [:wallet/valid-ens-or-address?]))
-            {:keys [color]}       (rf/sub [:wallet/current-viewing-account])]
+      (let [selected-tab             (or (rf/sub [:wallet/send-tab]) (:id (first tabs-data)))
+            valid-ens-or-address?    (boolean (rf/sub [:wallet/valid-ens-or-address?]))
+            local-suggestions        (rf/sub [:wallet/local-suggestions])
+            local-suggestion-address (:full-address (first local-suggestions))
```

Comment:
Just a suggestion: it's better to have a new sub that returns only what the view needs. In this case, any change to the output of `:wallet/local-suggestions` will force Reagent to re-process the entire component, but you only care about the `full-address` of the first suggestion. Maybe it's not a big deal here, but these details can easily pile up in terms of performance.

Not sure if it helps, but the mindset I put myself in is to **_not_** think of subs as normal functions. When we subscribe in the component, we're forming a dependency graph of subscriptions for the duration of the component lifecycle. This has a cost and affects re-renders, not like normal function calls that we can sprinkle anywhere and usually only care about what they do.

Reply:
Very good point. Done

Reply:
btw @mmilad75, this is a known problem we have throughout the codebase. I wrote a ton of code not following my own suggestion here, I guess because it took me a long time to understand the performance implications in mobile versus web development (which is where I came from using re-frame). Now we should gradually improve things or not make them worse if we can.